src/include/distributed/resource_lock.h

```diff
@@ -110,6 +111,15 @@ typedef enum CitusOperations
 						 ADV_LOCKTAG_CLASS_CITUS_PLACEMENT_CLEANUP)
 
 
+#define DIST_LOCK_DEFAULT 0
+#define DIST_LOCK_VIEWS_RECUR 1 << 0
+#define DIST_LOCK_REFERENCING_TABLES 1 << 1
+#define DIST_LOCK_NOWAIT 1 << 2
+
+#define LOCK_RELATION_IF_EXISTS "SELECT pg_catalog.lock_relation_if_exists(%s, '%s');"
+#define LOCK_RELATION_IF_EXISTS_NOWAIT \
+	"SELECT pg_catalog.lock_relation_if_exists(%s, '%s', nowait => true);"
```

Comment:
single quotes in a format string usually indicates a SQL injection bug.

probably not in this case because we know we always input a fixed string, but it would be good to use quote_literal_cstr() instead.